wordpress-mobile/WordPress-iOS
# [JP Plugin] Install Overlay UI

Fixes #20016 

This PR only adds the view and has no access point just yet. So testing is only possible with the test code provided below.

To test:
Replace the `presentOverlayIfNeeded` function in `JetpackFeaturesRemovalCoordinator.swift` as follows:

```Swift
    static func presentOverlayIfNeeded(in viewController: UIViewController,
                                       source: OverlaySource,
                                       forced: Bool = false,
                                       fullScreen: Bool = false,
                                       blog: Blog? = nil,
                                       onWillDismiss: JetpackOverlayDismissCallback? = nil,
                                       onDidDismiss: JetpackOverlayDismissCallback? = nil) {
        let phase = generalPhase()
        let frequencyConfig = phase.frequencyConfig
        let frequencyTrackerPhaseString = source.frequencyTrackerPhaseString(phase: phase)
//        var viewModel = JetpackFullscreenOverlayGeneralViewModel(phase: phase, source: source, blog: blog)
        var viewModel = JetpackPluginOverlayViewModel(siteName: "https://test.wordpress.com")
        viewModel.onWillDismiss = onWillDismiss
        viewModel.onDidDismiss = onDidDismiss
        let frequencyTracker = JetpackOverlayFrequencyTracker(frequencyConfig: frequencyConfig,
                                                              phaseString: frequencyTrackerPhaseString,
                                                              source: source)
//        guard viewModel.shouldShowOverlay, frequencyTracker.shouldShow(forced: forced) else {
//            onWillDismiss?()
//            onDidDismiss?()
//            return
//        }
        createAndPresentOverlay(with: viewModel, in: viewController, fullScreen: fullScreen)
        frequencyTracker.track()
    }
```

1. Launch Jetpack app
2. The modal should appear. Verify if the design matches and string attributes look meaningful.
3. Verify if the close button, closes the dialog

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
